src/js/tools/type.js

```diff
@@ -124,7 +126,15 @@ define(function (require, exports, module) {
                 var documentStore = this.flux.store("application"),
                     currentDocument = documentStore.getCurrentDocument();
 
-                this.flux.actions.layers.resetBounds(currentDocument, currentDocument.layers.allSelected);
+                // Debounce to allow history state event to precede the call to resetBounds
+                var debouncedMoveHandler = synchronization.debounce(function () {
```

Comment:
Yes, this is scary, and I am planning to investigate a better, more general solution to this problem of ordering history state events in the "PS initiated" flow.  But, this is a technique used elsewhere so is at least somewhat consistent.  for now.